PluginError.get() return error with no traceback

## 🐛 Bug

<!-- A clear and concise description of what the bug is. -->

I locally have the following error (code slightly edited for readability) :
```
napari/_qt/dialogs/_tests/test_qt_plugin_report.py F.                                [100%]

=========== FAILURES ====
______________________________________________________________ test_error_reporter ______________________________________________________________
CALL ERROR: Exceptions caught in Qt event loop:
________________________________________________________________________________
Traceback (most recent call last):
  File "/Users/mbussonnier/dev/napari/napari/_qt/dialogs/qt_plugin_report.py", line 162, in set_plugin
    err_string = format_exceptions(plugin, as_html=True)
  File "/Users/mbussonnier/dev/napari/napari/plugins/exceptions.py", line 72, in format_exceptions
    msg.append(format_exc_info(err.info(), as_html))
  File "/Users/mbussonnier/dev/napari/napari/utils/_tracebacks.py", line 44, in format_exc_info
    ansi_string = vbtb.text(etype, value, tb).replace(
  File "/Users/mbussonnier/dev/ipython/IPython/core/ultratb.py", line 285, in text
    tb_list = self.structured_traceback(etype, value, tb, tb_offset, context)
  File "/Users/mbussonnier/dev/ipython/IPython/core/ultratb.py", line 758, in structured_traceback
    formatted_exception = self.format_exception_as_a_whole(etype, evalue, etb, number_of_lines_of_context,
  File "/Users/mbussonnier/dev/ipython/IPython/core/ultratb.py", line 699, in format_exception_as_a_whole
    records = self.get_records(etb, number_of_lines_of_context, tb_offset)
  File "/Users/mbussonnier/dev/ipython/IPython/core/ultratb.py", line 751, in get_records
    res = list(stack_data.FrameInfo.stack_data(etb, options=options))[tb_offset:]
  File "/Users/mbussonnier/miniconda3/envs/napari/lib/python3.9/site-packages/stack_data/core.py", line 527, in stack_data
    if is_frame(frame_or_tb):
  File "/Users/mbussonnier/miniconda3/envs/napari/lib/python3.9/site-packages/stack_data/utils.py", line 87, in is_frame
    assert isinstance(frame_or_tb, (types.FrameType, types.TracebackType)), frame_or_tb
AssertionError: None
```


Which I tracked down to 
```
 _plugin_errors = PluginError.get(plugin_name=plugin_name)
...
for n, err in enumerate(_plugin_errors):
        ...
        # _, _, tb = err.info()
        # assert tb is not None. # would fail. Need to be
        msg.append(format_exc_info(err.info(), ....))
        ...
```

I have issue tracking it down further.

--- a/napari/_qt/dialogs/_tests/test_qt_plugin_report.py
+++ b/napari/_qt/dialogs/_tests/test_qt_plugin_report.py
@@ -21,7 +21,13 @@ def test_error_reporter(qtbot, monkeypatch):
     )
 
     error_message = 'my special error'
-    _ = PluginError(error_message, plugin_name='test_plugin', plugin="mock")
+    try:
+        # we need to raise to make sure a __traceback__ is attached to the error.
+        raise PluginError(
+            error_message, plugin_name='test_plugin', plugin="mock"
+        )
+    except PluginError:
+        pass
     report_widget = qt_plugin_report.QtPluginErrReporter()
     qtbot.addWidget(report_widget)
 

